src/daft-core/src/array/ops/approx_count_distinct_merge.rs

```diff
@@ -0,0 +1,46 @@
+use arrow2::array::PrimitiveArray;
+use common_error::DaftResult;
+
+use crate::{
+    array::ops::{as_arrow::AsArrow, DaftApproxCountDistinctMergeAggable},
+    datatypes::{Field, FixedSizeBinaryArray, UInt64Array},
+    utils::hyperloglog::HyperLogLog,
+    DataType,
+};
+
+use crate::array::ops::GroupIndices;
+
+impl DaftApproxCountDistinctMergeAggable for &FixedSizeBinaryArray {
+    type Output = DaftResult<UInt64Array>;
+
+    fn approx_count_distinct_merge(&self) -> Self::Output {
+        let mut final_hll = HyperLogLog::default();
+        for byte_slice in self.as_arrow().values_iter() {
+            let hll = HyperLogLog::new_with_byte_slice(byte_slice);
```

Comment:
We should look into if we can make an HLL sketch that leverages a borrowed slice

Reply:
That would actually be a performant solution. The copy here is eating up time.

Potentially will look into updating the internals of HLL from just a simple `[u8; NUM_REGISTERS]` to an `AsRef<'a, [u8; NUM_REGISTERS]>` instead.

That way, we could instantiate the `HyperLogLog` struct with a simple mutable reference instead.

Reply:
Updated!